frontend/src/store.ts

```diff
@@ -1,18 +1,62 @@
-import { configureStore } from '@reduxjs/toolkit'
+import { ThunkMiddleware, configureStore } from '@reduxjs/toolkit'
 import { createLogger } from 'redux-logger'
 import rootReducer from './reducers'
+import { showNotification } from './modules/notification/actions'
+import { logOut } from './modules/auth/actions'
+import shouldIgnoreError from './utils/shouldIgnoreError'
 
 const logger = createLogger({
 	level: 'info'
 })
 
+const notificationError: ThunkMiddleware = ({ dispatch }) => next => action => {
+	const TOKEN_EXPIRED_MESSAGE = 'Given token not valid for any token type'
+
+	function getErrorDescription(error: any) {
+		/* HTTP errors are handled specially because they include the URL
+		 * in the error message and we don't want many very similar errors
+		 * to show up, just show one.
+		 */
+		if (typeof error.status === 'number')
```

Comment:
I think for network errors it would be better if we use the URL as the notification ID, rather than the status code. For example, we might get 500 errors from multiple endpoints, and I think it would make sense to show a separate notification for each failed request, rather than replacing the alert each time a 500 occurs.